woocommerce/woocommerce-ios
# Add support for storing multiple feedback states

relates to https://github.com/woocommerce/woocommerce-ios/issues/2647

# Why

The app should support multiple feedback requests. At the time of this writing, a general feedback is requested and we are close to asking feedback for the [product feature](https://github.com/woocommerce/woocommerce-ios/issues/2648).

This PR aims to extend the `GeneralAppSettings` type to save multiple feedback states, one for each feedback.

# How

- Defined a new `FeedbackSettings` type that stores a particular feedback state. Our current options for that state are `pending`, `dismissed` or `given`.
- Modify `GeneralAppSettings` to include this new feedback type
- Modify `AppSettingsStore` to save the `last feedback date` using the new `FeedbackSettings` type.

# Testing Steps
- See that unit tests pass

And

- Delete the app on the device.
- Run the app and log in.
- Time travel to at least 90 days from now.
- Navigate to the Orders tab and navigate back to My Store.
- Confirm that the in-app feedback card is now visible.
- Tap on any CTA
- See that the in-app feedback is hidden
- Relaunch the app
- See that the in-app feedback remains invisible

# Notes
- The test that is failing should be fixed once https://github.com/woocommerce/woocommerce-ios/pull/2656 is merged

Update release notes:
